server/utils.go

```diff
@@ -205,7 +205,7 @@ func filterLines(s string, start, end int) (string, error) {
 		buf.WriteByte(byte('\n'))
 	}
 
-	return buf.String(), scanner.Err()
+	return buf.String(), errors.Wrap(scanner.Err(), "scanner error occurred")
```

Comment:
That error handling is really weird. Given that you are touching the code anyway, would you mind fixing the error handling and doing a standard `if err := scanner.Err(); err != nil { ...`

Reply:
sure

Reply:
Since I am already working with errors, I took it upon myself to handle some more unhandled errors. Also, there are some parts of the code (particularly in plugin.go) where an error can occur but the function doesn't return errors so it can't be handled, I figured I would at least log those errors.  Is it ok if I make these changes?